Server/controllers/authController.js

```diff
@@ -234,11 +235,67 @@ const resetPasswordController = async (req, res, next) => {
     next(error);
   }
 };
+
+/**
+ * Deletes a user and all associated monitors, checks, and alerts.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Object} The response object with success status and message.
+ * @throws {Error} If user validation fails or user is not found in the database.
+ */
+const deleteUserController = async (req, res, next) => {
+  try {
+    // Validate user
+    await userDeleteValidation.validateAsync(req.params.userId);
```

Comment:
Here you are passing a string to the Joi validator, however the validator you defined:

```
const userDeleteValidation = joi.object({
  userId: joi.string().required(),
});
```
expects an object with the a key of `userId`. 

You should pass it the params of the request instead of the userId to comply with your definition:

 `await userDeleteValidation.validateAsync(req.params);`